packages/blockchain-wallet-v4-frontend/src/data/coins/selectors/coins/btc.tsx

```diff
@@ -37,42 +37,52 @@ export const getAccounts = createDeepEqualSelector(
     ) => {
       const { coin } = ownProps
       let accounts = []
-
       // add non-custodial accounts if requested
       if (ownProps?.nonCustodialAccounts) {
-        accounts = accounts.concat(btcAccounts
-          .map(acc => ({
-            accountIndex: prop('index', acc),
-            address: prop('index', acc),
-            archived: prop('archived', acc),
-            // @ts-ignore
-            balance: prop('final_balance', prop(prop('xpub', acc), btcData)),
-            baseCoin: coin,
-            coin,
-            label: prop('label', acc) || prop('xpub', acc),
-            type: ADDRESS_TYPES.ACCOUNT
-          }))
-          .filter(propEq('archived', false)))
+        // each account has a derivations object with legacy address and segwit address
+        // need to extract each xpub for balance
+        const xpubArray = acc =>
```

Comment:
gnarly logic begins here. If there's a simpler way to do this, I'd love it.

Reply:
there might be a generic selector for doing this. if not we should consider making one. looking now 👀

Reply:
actually this is fine the way it is